src/TransferInformation/BaseTransferInformation.php

```diff
@@ -104,19 +104,14 @@ class BaseTransferInformation implements TransferInformationInterface
      * @param string $amount
      * @param string $iban
      * @param string $name
-     *
-     * @throws InvalidArgumentException
      */
     public function __construct($amount, $iban, $name)
     {
-        $amount += 0;
-        if (is_float($amount)) {
-            if (!function_exists('bcscale')) {
-                throw new InvalidArgumentException('Using floats for amount is only possible with bcmath enabled');
-            }
-            bcscale(2);
-            $amount = (integer)bcmul(sprintf('%01.4F', $amount), '100');
+        if(preg_match("#^([0-9]+)\.([0-9]+)$#", $amount, $matches)) {
```

Comment:
We need to implement some test for this. This is a huge change in my opinion and it should be safeguarded by some tests